src/compiler/checker.ts

```diff
@@ -6743,15 +6775,28 @@ module ts {
                 if (arg.kind !== SyntaxKind.OmittedExpression) {
                     let paramType = getTypeAtPosition(signature, i);
                     let argType: Type;
-                    if (i === 0 && args[i].parent.kind === SyntaxKind.TaggedTemplateExpression) {
+                    if (arg.parent.kind === SyntaxKind.Decorator) {

```

Comment:
I think `arg` is a bit of a misnomer anyway. This thing is not the argument, it's the target. That's why this whole effective arguments thing makes me uncomfortable